kfs-core/src/main/java/edu/arizona/kfs/pdp/batch/AchBankInputFileType.java

```diff
@@ -0,0 +1,371 @@
+			reportLines.add(new BankReportLine(bankRoutingNbr, bankNm, KFSConstants.ACHFileConstants.ACH_BANK_INVALID_INST_STAT_CD));
+		}
+	
+	
+		if (StringUtils.isBlank(bankDataViewCd)) {
+			bank.setBankDataViewCode(KFSConstants.ACHFileDefaultConstants.DEFAULT_BANK_DATA_VIEW_CODE);
+			reportLines.add(new BankReportLine(bankRoutingNbr, bankNm, KFSConstants.ACHFileConstants.ACH_BANK_INVALID_DATA_VIEW_CD));
+		}
+		
+		return true;
+	}
+		
+	protected void writeReport(List<BankReportLine> lines, String systemSource) {
+		File reportFile = new File(reportPath + "/" + reportPrefix + "_" + dateTimeService.toDateTimeStringForFilename(dateTimeService.getCurrentDate()) + "." + reportExtension);
+		BufferedWriter writer = null;
+		String routingNbr;
+		String bankNm;
+		
+		try {
+			writer = new BufferedWriter(new PrintWriter(reportFile));
+			writeReportTitle(writer, systemSource);
+			writeReportHeadings(writer);
+			for (BankReportLine line : lines) {
+				routingNbr = rightPad(line.getBankRoutingNumber(), 15, " ");
+				bankNm = rightPad(line.getBankName(), 40, " ");
+				writer.write(routingNbr + bankNm + line.getMessage());
+				writer.newLine();
+				writer.newLine();
+			}
+		}
+		catch (FileNotFoundException e) {
+			LOG.error(reportFile + " not found " + " " + e.getMessage());
+			throw new RuntimeException(reportFile + " not found" + e.getMessage(), e);
+		}
+		catch (IOException e) {
+			LOG.error("Error writing to BufferedWriter " + e.getMessage());
+			throw new RuntimeException("Error writing to BufferedWriter " + e.getMessage(), e);
+		}
+		finally {
+			try {
+				writer.flush();
+				writer.close();
+			}
+			catch (Exception e) {
+			}
+		}
+	}
+	
+	protected void writeReportTitle(BufferedWriter writer, String systemSource) {
+		OriginationCode originCd = originationCodeDao.findByCode(systemSource);
+		String source = systemSource + " - Unknown System Source";
+		if (originCd != null) {
+			source = originCd.getFinancialSystemDatabaseDesc();
+		}
+		try {
+			writer.write(dateTimeService.toDateString(dateTimeService.getCurrentDate()));
+			writer.newLine();
+			writer.newLine();
+			writer.write("ACH Bank Interface");
+			writer.newLine();
+			writer.write("System Source: " + systemSource + " - " + source);
+			writer.newLine();
+			writer.newLine();
+		}
+		catch(IOException e) {
+			LOG.error("Error writing to BufferedWriter " + e.getMessage());
+			throw new RuntimeException("Error writing to BufferedWriter " + e.getMessage(), e);
+		}
+	}
+	
+	protected void writeReportHeadings(BufferedWriter writer) {
+		String bankRoutingHdr = rightPad("BankRouting #", 15, " ");
+		String bankNmHdr = rightPad("Bank Name", 40, " ");
+		
+		try {
+			writer.write(bankRoutingHdr + bankNmHdr + "Message");
+			writer.newLine();
+			writer.newLine();
+		}
+		catch (IOException e) {
+			LOG.error("Error writing to BufferedWriter " + e.getMessage());
+			throw new RuntimeException("Error writing to BufferedWriter " + e.getMessage(), e);
+		}
```

Comment:
I suppose I should have paid better attention where these were being "newed" up ;). Thanks for catching these instances.